src/lib/core/components/visualizations/implementations/MapVisualization.tsx

```diff
@@ -0,0 +1,242 @@
+          },
+        });
+      }
+    },
+    [updateVizConfig]
+  );
+
+  const tempOutputEntityId = entities[0].id;
+
+  const [boundsZoomLevel, setBoundsZoomLevel] = useState<BoundsViewport>();
+
+  const latitudeVariableDetails = useMemo(
+    () => ({
+      entityId: tempOutputEntityId,
+      variableId: 'OBI_0001620',
+    }),
+    [tempOutputEntityId]
+  );
+  const longitudeVariableDetails = useMemo(
+    () => ({
+      entityId: tempOutputEntityId,
+      variableId: 'OBI_0001621',
+    }),
+    [tempOutputEntityId]
+  );
+
+  const markers = usePromise<
+    Array<ReactElement<BoundsDriftMarkerProps>> | undefined
+  >(
+    useCallback(async () => {
+      if (boundsZoomLevel == null) return [];
+
+      const { bounds, zoomLevel } = boundsZoomLevel;
+
+      const boundsFilters = filtersFromBoundingBox(
+        bounds,
+        latitudeVariableDetails,
+        longitudeVariableDetails
+      );
+      // TO DO: add bounding box-based filters!
+
+      const requestParams: MapMarkersRequestParams = {
+        studyId,
+        filters: filters ? [...filters, ...boundsFilters] : boundsFilters,
+        config: {
+          outputEntityId: tempOutputEntityId,
+          geoAggregateVariable: {
+            entityId: tempOutputEntityId,
+            variableId: geohashLevelToVariableId(
+              leafletZoomLevelToGeohashLevel(zoomLevel)
+            ),
+          },
+          latitudeVariable: latitudeVariableDetails,
+          longitudeVariable: longitudeVariableDetails,
+        },
+      };
+      const response = await dataClient.getMapMarkers(
+        computation.descriptor.type,
+        requestParams
+      );
+
+      // TO DO: find out if MarkerProps.id is obsolete
+      return response.mapElements.map(
+        ({
+          avgLat,
+          avgLon,
+          minLat,
+          minLon,
+          maxLat,
+          maxLon,
+          entityCount,
+          geoAggregateValue,
+        }) => {
+          const isAtomic = false; // TO DO: work with Danielle to get this info from back end
+          const data = [
+            {
+              label: 'unknown',
+              value: entityCount,
+              color: 'white',
+            },
+          ];
+          return (
+            <DonutMarker
+              id={geoAggregateValue}
+              key={geoAggregateValue}
+              position={{ lat: avgLat, lng: avgLon }}
+              bounds={{
+                southWest: { lat: minLat, lng: minLon },
+                northEast: { lat: maxLat, lng: maxLon },
+              }}
+              data={data}
+              isAtomic={isAtomic}
+              duration={defaultAnimationDuration}
+            />
+          );
+        }
+      );
+    }, [
+      studyId,
+      filters,
+      dataClient,
+      tempOutputEntityId,
+      latitudeVariableDetails,
+      longitudeVariableDetails,
+      boundsZoomLevel,
+      computation.descriptor.type,
+      defaultAnimationDuration,
```

Comment:
```suggestion
```

Clearing another eslint warning for unnecessary dependency

Reply:
`defaultAnimationDuration` is passed as a prop to the `DonutMarker` component, so I think it's needed?

Reply:
While it is used in the callback, I believe the idea is that `defaultAnimationDuration` is a module import and so it won't ever change in the component's lifecycle